### PR TITLE
fix: used the Timeout instead of two timeoutHeight and timeoutTimestamp 

### PR DESCRIPTION
**Pull request by vishal-kanna**
## Description

closes: #3727 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: testing/endpoint.go [hunk 1]
```diff
@@ -431,7 +431,7 @@ func (endpoint *Endpoint) SendPacket(
 	channelCap := endpoint.Chain.GetChannelCapability(endpoint.ChannelConfig.PortID, endpoint.ChannelID)
```
Comment (charleenfei) [inline testing/endpoint.go:431]: I think you can also update the SendPacket api on this endpoint :)

File: modules/apps/29-fee/ibc_middleware.go [hunk 2]
```diff
@@ -326,11 +325,10 @@ func (im IBCMiddleware) SendPacket(
 	chanCap *capabilitytypes.Capability,
 	sourcePort string,
 	sourceChannel string,
-	timeoutHeight clienttypes.Height,
-	timeoutTimestamp uint64,
+	timeout channeltypes.Timeout,
 	data []byte,
 ) (uint64, error) {
-	return im.keeper.SendPacket(ctx, chanCap, sourcePort, sourceChannel, timeoutHeight, timeoutTimestamp, data)
+	return im.keeper.SendPacket(ctx, chanCap, sourcePort, sourceChannel, channeltypes.Timeout{Height: timeout.Height, Timestamp: timeout.Timestamp}, data)
```
Comment (charleenfei) [inline modules/apps/29-fee/ibc_middleware.go:331]: why not use the `timeout` param passed in directly?

File: modules/apps/29-fee/keeper/relay.go [hunk 3]
```diff
@@ -19,11 +18,10 @@ func (k Keeper) SendPacket(
 	chanCap *capabilitytypes.Capability,
 	sourcePort string,
 	sourceChannel string,
-	timeoutHeight clienttypes.Height,
-	timeoutTimestamp uint64,
+	timeout channeltypes.Timeout,
 	data []byte,
 ) (uint64, error) {
-	return k.ics4Wrapper.SendPacket(ctx, chanCap, sourcePort, sourceChannel, timeoutHeight, timeoutTimestamp, data)
+	return k.ics4Wrapper.SendPacket(ctx, chanCap, sourcePort, sourceChannel, channeltypes.Timeout{Height: timeout.Height, Timestamp: timeout.Timestamp}, data)
```
Comment (charleenfei) [inline modules/apps/29-fee/keeper/relay.go:24]: same comment as above

File: modules/apps/transfer/keeper/relay.go [hunk 4]
```diff
@@ -131,7 +131,7 @@ func (k Keeper) sendTransfer(
 		fullDenomPath, token.Amount.String(), sender.String(), receiver, memo,
 	)
 
-	sequence, err := k.ics4Wrapper.SendPacket(ctx, channelCap, sourcePort, sourceChannel, timeoutHeight, timeoutTimestamp, packetData.GetBytes())
+	sequence, err := k.ics4Wrapper.SendPacket(ctx, channelCap, sourcePort, sourceChannel, channeltypes.Timeout{Height: timeoutHeight, Timestamp: timeoutTimestamp}, packetData.GetBytes())
```
Comment (charleenfei) [inline modules/apps/transfer/keeper/relay.go:134]: same

**Comment by crodriguezvega** [comment]
@vishal-kanna, thank you very much for opening this PR. However, we discussed this internally and as @colin-axner explains in [his comment](https://github.com/cosmos/ibc-go/issues/3727#issuecomment-1580498753), it might be better to hold on implementing this change for now since it will break IBC applications. So I hope you don't mint, but I will close the PR now. Again, thank you very much for taking the initiative to implement this!

**Pull request closed without merging**